### PR TITLE
Don't run store migrations on renderer

### LINKED ISSUE #3366: Black screen on startup
**Describe the bug**

I restarted Lens and window is now stuck on black screen (nothing is showing up in the window). It worked well yesterday and for many weeks. I used v5.0.2. I did try to reinstaller but it didn't fix the issue.

**Expected behavior**
Lens is loading

**Screenshots**

Screenshot 2021-07-13 at 09 53 03

**Environment (please complete the following information):**
- Lens Version: 5.0.2
- OS: macOS 11.4
- Installation method (e.g. snap or AppImage in Linux): Homebrew

**Logs:**

```
 ~  /Applications/Lens.app/Contents/MacOS/Lens ; exit;             ✔  09:47:49 
info: 📟 Setting Lens as protocol client for lens://
info: 📟 Protocol client register succeeded ✅
info: 🚀 Starting Lens from "/Users/example/Library/Application Support/Lens"
info: 🐚 Syncing shell environment
info: 💾 Loading stores
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/lens-user-store.json
2021-07-13 09:47:55.478 Lens[38864:1210137] FinderSync Collaboration: remote proxy error = Error Domain=NSCocoaErrorDomain Code=4099 "The connection on anonymousListener or serviceListener from pid 39252 was invalidated." UserInfo={NSDebugDescription=The connection on anonymousListener or serviceListener from pid 39252 was invalidated.}
2021-07-13 09:47:59.404 Lens[38864:1210010] FinderSync Collaboration: remote proxy error = Error Domain=NSCocoaErrorDomain Code=4099 "The connection on anonymousListener or serviceListener from pid 39252 was invalidated." UserInfo={NSDebugDescription=The connection on anonymousListener or serviceListener from pid 39252 was invalidated.}
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/lens-cluster-store.json
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/lens-hotbar-store.json
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/lens-extensions.json
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/lens-weblink-store.json
info: 🔌 Starting LensProxy
info: [LENS-PROXY]: Proxy server has started at 127.0.0.1:52229
info: 🔎 Testing LensProxy connection ...
info: ⚡ LensProxy connection OK
2021-07-13 09:48:04.776 Lens[38864:1210154] FinderSync Collaboration: remote proxy error = Error Domain=NSCocoaErrorDomain Code=4099 "The connection on anonymousListener or serviceListener from pid 39252 was invalidated." UserInfo={NSDebugDescription=The connection on anonymousListener or serviceListener from pid 39252 was invalidated.}
2021-07-13 09:48:09.778 Lens[38864:1210038] FinderSync Collaboration: remote proxy error = Error Domain=NSCocoaErrorDomain Code=4099 "The connection on anonymousListener or serviceListener from pid 39252 was invalidated." UserInfo={NSDebugDescription=The connection on anonymousListener or serviceListener from pid 39252 was invalidated.}
info: 🖥️  Starting WindowManager
info: 🧩 Initializing extensions
info: [EXTENSION-DISCOVERY] loading extensions from /Users/example/Library/Application Support/Lens
info: APP:ACTIVATE {"hasVisibleWindows":false}
info: [WINDOW-MANAGER]: Main window has reported that it has loaded
info: [EXTENSION-INSTALLER] installing dependencies at /Users/example/Library/Application Support/Lens
info: [EXTENSION-INSTALLER] dependencies installed at /Users/example/Library/Application Support/Lens
info: [EXTENSION-DISCOVERY] watching extension add/remove in /Users/example/.k8slens/extensions
Native extraction module not loaded, msgpackr will still run, but with decreased performance. Cannot find module 'msgpackr-extract'
Lens Cloud Extension activated
info: [STORE]: LOADED from /Users/example/Library/Application Support/Lens/extension-store/lenscloud-lens-extension/auth-token-store.json
error: [EXTENSIONS-LOADER]: can't load main for "lens-telemetry": TypeError: Cannot create a second singleton while creating a first {"extension":{"id":"/Users/example/Library/Application Support/Lens/node_modules/lens-telemetry/package.json","absolutePath":"/Applications/Lens.app/Contents/Resources/extensions/telemetry/lens-telemetry-5.0.2-latest.20210705.2.tgz","manifestPath":"/Users/example/Library/Application Support/Lens/node_modules/lens-telemetry/package.json","manifest":{"name":"lens-telemetry","version":"5.0.2-latest.20210705.2","description":"Lens IDE telemetry","main":"dist/main.js","renderer":"dist/renderer.js","lens":{"metadata":{},"styles":[]},"scripts":{"build":"webpack -p && npm pack","dev":"webpack --watch","test":"jest --passWithNoTests --env=jsdom src $@"},"files":["dist/**/*"],"devDependencies":{"@k8slens/extensions":"file:../../src/extensions/npm/extensions","@types/analytics-node":"^3.1.3","@types/node":"^14.14.14","@types/react":"^17.0.0","@types/universal-analytics":"^0.4.4","analytics-node":"^3.4.0-beta.3","conf":"^7.1.2","jest":"^26.6.3","mobx":"^6.3.2","mobx-react":"^7.1.0","node-machine-id":"^1.1.12","react":"^16.13.1","ts-loader":"^8.0.4","typescript":"^4.0.3","universal-analytics":"^0.4.23","webpack":"^4.44.2","webpack-cli":"^3.3.11"}},"isBundled":true,"isEnabled":true,"isCompatible":true}}
error: [EXTENSION]: failed to activate lens-survey@5.0.2-latest.20210705.2: Error: [MobX] No annotations were passed to makeObservable, but no decorated members have been found either
info: [EXTENSION]: enabled lens-license@5.0.2-latest.20210705.2
[CLUSTER-CONNECT] cluster proxy started on port 52239
[CLUSTER-CONNECT] cluster proxy is using aes-256-gcm for end-to-end encryption
info: [EXTENSION]: enabled lenscloud-lens-extension@5.0.2-latest.20210705.2
```

**Kubeconfig:**

Checked with a validator, it is valid

**Additional context**

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

Fixes #3366

### CODE REVIEW

File: src/migrations/user-store/5.0.3-beta.1.ts [hunk 1]
```diff
@@ -19,22 +19,21 @@
...
 
-import { app } from "electron";
 import { existsSync, readFileSync } from "fs";
 import path from "path";
 import os from "os";
 import type { ClusterStoreModel } from "../../common/cluster-store";
 import type { KubeconfigSyncEntry, UserPreferencesModel } from "../../common/user-store";
 import { MigrationDeclaration, migrationLog } from "../helpers";
-import { isLogicalChildPath, storedKubeConfigFolder } from "../../common/utils";
+import { getPath, isLogicalChildPath, storedKubeConfigFolder } from "../../common/utils";
```
Comment (jweak) [inline src/migrations/user-store/5.0.3-beta.1.ts]: Isn't getPath `@deprecated` ?

**Comment by jakolehm** [comment]
@Nokel81 does this mean that migrations are also executed in renderer? (that does not sound right)

File: src/migrations/user-store/5.0.3-beta.1.ts (hunk 1 above)
Comment (Nokel81) [reply src/migrations/user-store/5.0.3-beta.1.ts]: Yes because we should be trying to remove uses of `remote`

**Comment by Nokel81** [comment]
@jakolehm Apparently? It seems very strange, but the error in the linked issue is most certainly a migration error. 

I agree that it is wrong. Perhaps I should come up with a better solution.

**Comment by Nokel81** [comment]
@jakolehm PTAL now?

File: src/migrations/user-store/5.0.3-beta.1.ts (hunk 1 above)
Comment (jakolehm) [reply src/migrations/user-store/5.0.3-beta.1.ts]: Do we have alternative to `getPath`? If no -> it feels weird.

**Review by jakolehm — APPROVED** [review]

**Pull request merged**